### PR TITLE
Update PSAT requirements documentation 

### PR DESCRIPTION
**Pull request by cccsss01**

### CODE REVIEW

**Review by kfox1111 — APPROVED** [review]
Very nice. Thanks for the fix. :)

LGTM

**Comment by cccsss01** [comment]
Well, not sure if it's really needed since it requires 1.22 anyways.

File: charts/spire/README.md [hunk 1]
```diff
@@ -27,6 +27,12 @@ A Helm chart for deploying the complete Spire stack including: spire-server, spi
 
 Please note this chart requires `Projected Service Account Tokens` which has to be enabled on your k8s api server.
 
+> **Note** This is enabled by default with newer versions as shown by the existence of
+        - --service-account-issuer
+        - --service-account-key-file
+        - --service-account-signing-key-file
+See https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#serviceaccount-token-volume-projection for more details
```
Comment (faisal-memon) [inline charts/spire/README.md:34]:
```suggestion
See [Service Account Token Volume Projection](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#serviceaccount-token-volume-projection) in the Kubernetes docs for more details.
```

**Comment by faisal-memon** [comment]
> Well, not sure if it's really needed since it requires 1.22 anyways.

@cccsss01 I think you are right. We should probably just remove that whole section to avoid confusion.

**Comment by marcofranssen** [comment]
Change has to be made in https://github.com/spiffe/helm-charts/blob/main/charts/spire/README.md.gotmpl

so when we run helm-docs.sh the README.md is updated.

That is the reason for failing build.

**Comment by faisal-memon** [comment]
@cccsss01 What do you want to do here?

**Comment by marcofranssen** [comment]
We have created a new PR #437 which will move this information to a more clear location for those few usecases where people can't get PSAT to work.

This way we don't confuse the majority of users with instructions that are only required on very old k8s deployments.

**Comment by faisal-memon** [comment]
Will be covered as part of #437 . Thanks @cccsss01 for bring this to our attention.

**Pull request closed without merging**